src/main/java/com/github/chabanenk0/Service/BeanComparator.java

```diff
@@ -0,0 +1,51 @@
+package com.github.chabanenk0.Service;
+
+import com.github.chabanenk0.Annotation.Compare;
+import com.github.chabanenk0.Entity.BeanComparator.CompareResult;
+import java.lang.Object;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.security.InvalidParameterException;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Created by dmitry on 21.12.16.
+ */
+public class BeanComparator
+{
+    public List<CompareResult> compare(Object object1, Object object2)
+    {
+        List<CompareResult> results = new LinkedList<CompareResult>();
+        Class object1Class = object1.getClass();
+        Class object2Class = object2.getClass();
+
+        if (object1Class != object2Class) {
+            throw new InvalidParameterException("Cannot compare objects");
+        }
+
+        for(Field field: object1Class.getDeclaredFields()) {
+            if (field.isAnnotationPresent(Compare.class)) {
```

Comment:
if (!field.isAnnotationPresent(Compare.class)) continue;
as an alternative, 1 less level of nesting